[SofaMatrix] Allow newer metis versions

LunNova opened the pull request:
Fixes #5697

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

fredroy commented:
The requirement of having an exact version for metis was made on purpose apparently, according to @alxbilger  https://github.com/sofa-framework/sofa/pull/4450#issuecomment-1905527144

LunNova commented:
Can it be relaxed to allow >=5.2.1 or 5.1.0 exactly then? Sounds like there was a faulty in between version that must be avoided.

5.1.0 is from 2013 and incompatible with CMake 4.

alxbilger commented:
I think at the time we forced the exact version, the version 5.2.1 was not available on conda. That is why we forced version 5.1.0. Based on https://github.com/conda-forge/metis-feedstock/pull/43, it seems it is now available (@olivier-roussel do you confirm?). In that case, it worth trying to apply your suggestion (>=5.2.1 or 5.1.0 exactly). You can try and see if the CI is happy.

LunNova committed:
[SofaMatrix] Allow metis >= 5.2.1

Fixes #5697

metis 5.1.0 is known to work, versions newer than 5.1 and older than
5.2.1 are broken. A fix was applied in a commit[1] that was first
included in 5.2.1.

[1]: https://github.com/KarypisLab/METIS/commit/36262adecaa9720a4417a67124428061c367fd3f

LunNova commented:
ubuntu check failure was `g++: fatal error: Killed signal terminated program cc1plus`. I'd hope that's not related to this, did that box OOM?

fredroy commented:
[ci-build][with-all-tests]

olivier-roussel commented:
Yes, we fixed metis 5.1.0 initially as the latest release at this time 5.1.1 was apparently broken ( https://github.com/KarypisLab/METIS/issues/69, https://github.com/conda-forge/metis-feedstock/issues/35 ).
With the release 5.2.1 which is available also on conda-forge, if CI passes we should move to it.

alxbilger commented:
@LunNova The compilation issue on Ubuntu is not due to your changes.

alxbilger commented:
Note that if this PR is accepted, it will also make sense to apply it on https://github.com/sofa-framework/Sofa.Metis/blob/master/CMakeLists.txt

alxbilger commented:
@bakpaul Do you see a way to run the CI with Metis 5.2.1?

olivier-roussel commented:
To be more precise, we do not use so far metis conda package as we do not provide a conda package for Sofa.Metis plugin. The conda package use was at the time where Sofa.Metis was still part of SOFA. And from what I remember we were using a specific version of metis compiled with some extra options, which are typically not available from current conda binaries yet. We can still dig about that if at some point we want of conda package for Sofa.Metis, maybe this can be achieved by pushing for a binary variant of the package with these options, if that make sense for the community.
This remark does not apply with Pixi build chain which fetches metis.

bakpaul commented:
@alxbilger I just need to fetch the branch on our METIS fork, rework the packaging again and we can try it with a simple ci-depends-on, but not on this PR, I'll open a drafted one to test it so we can merge this asap.

bakpaul commented:
Well actually it seems like we have already tried this version, and I don't remember if we dropped it because of Conda constraints or if it was because it was messing with the unit tests. Let's see : https://github.com/sofa-framework/sofa/pull/5703

fredroy commented:
Closing, #5732 has been merged and is based on this one
